src/test/enhetstester/kotlin/no/nav/familie/ks/sak/sikkerhet/TilgangServiceTest.kt

```diff
@@ -0,0 +1,502 @@
+    }
+
+    @Test
+    internal fun `skal kaste RolleTilgangskontrollFeil dersom bruker verken er veileder saksbehandler eller beslutter`() {
+        mockBrukerContext(groups = emptyList())
+
+        assertThrows<RolleTilgangskontrollFeil> {
+            tilgangService.validerTilgangTilHandling(BehandlerRolle.VEILEDER, "")
+        }
+    }
+
+    @Test
+    internal fun `skal ikke kaste RolleTilgangskontrollFeil dersom beslutter beslutter forsøker å gjøre handling som krever BESLUTTER-rolle`() {
+        mockBrukerContext("A", listOf(BehandlerRolle.BESLUTTER.name))
+        tilgangService.validerTilgangTilHandling(BehandlerRolle.BESLUTTER, "")
+    }
+
+    @Test
+    internal fun `skal ikke kaste RolleTilgangskontrollFeil dersom saksbehandler eller beslutter forsøker å gjøre handling som krever SAKSBEHANDLER-rolle`() {
+        mockBrukerContext("A", listOf(BehandlerRolle.BESLUTTER.name))
+        tilgangService.validerTilgangTilHandling(BehandlerRolle.SAKSBEHANDLER, "")
+
+        mockBrukerContext("A", listOf(BehandlerRolle.SAKSBEHANDLER.name))
+        tilgangService.validerTilgangTilHandling(BehandlerRolle.SAKSBEHANDLER, "")
+    }
+
+    @Test
+    internal fun `skal ikke kaste RolleTilgangskontrollFeil dersom saksbehandler beslutter eller veileder forsøker å gjøre handling som krever VEILEDER-rolle`() {
+        mockBrukerContext("A", listOf(BehandlerRolle.BESLUTTER.name))
+        tilgangService.validerTilgangTilHandling(BehandlerRolle.VEILEDER, "")
+
+        mockBrukerContext("A", listOf(BehandlerRolle.SAKSBEHANDLER.name))
+        tilgangService.validerTilgangTilHandling(BehandlerRolle.VEILEDER, "")
+
+        mockBrukerContext("A", listOf(BehandlerRolle.VEILEDER.name))
+        tilgangService.validerTilgangTilHandling(BehandlerRolle.VEILEDER, "")
+    }
+
+    @Test
+    internal fun `skal kaste RolleTilgangskontrollFeil dersom saksbehandler ikke har tilgang til person`() {
+        every { integrasjonService.sjekkTilgangTilPersoner(any()) } returns Tilgang(false)
+
+        assertThrows<RolleTilgangskontrollFeil> {
+            tilgangService.validerTilgangTilPersoner(
+                listOf(aktør.aktivFødselsnummer()),
+                AuditLoggerEvent.ACCESS,
+                BehandlerRolle.SAKSBEHANDLER,
+                ""
+            )
+        }
+    }
+
+    @Test
+    internal fun `skal ikke feile når saksbehandler har tilgang til person`() {
+        every { integrasjonService.sjekkTilgangTilPersoner(any()) } returns Tilgang(true)
+
+        tilgangService.validerTilgangTilPersoner(
+            listOf(aktør.aktivFødselsnummer()),
+            AuditLoggerEvent.ACCESS,
+            BehandlerRolle.SAKSBEHANDLER,
+            ""
+        )
+    }
+
+    @Test
+    internal fun `skal kaste RolleTilgangskontrollFeil dersom saksbehandler ikke har tilgang til behandling`() {
+        every { integrasjonService.sjekkTilgangTilPersoner(any()) } returns Tilgang(false)
+
+        assertThrows<RolleTilgangskontrollFeil> {
```

Comment:
Det kan kanskje være greit å asserte på meldingen som kastes her, bare for å validere at det er riktig grunn til at en `RolleTilgangskontrollFeil` kastes